From 51e81c2572df82bc70bda151cfa893b6497d3f54 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:06 -0400
Subject: [usb] wusbcore: change WA_SEGS_MAX to a legal value

Message-id: <1410887463-20674-115-git-send-email-dzickus@redhat.com>
Patchwork-id: 95122
O-Subject: [RHEL7 PATCH 114/591] usb: wusbcore: change WA_SEGS_MAX to a legal value
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f74b75e7f920c700636cccca669c7d16d12e9202
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Wed Oct 23 14:44:29 2013 -0500

    usb: wusbcore: change WA_SEGS_MAX to a legal value

    change WA_SEGS_MAX to a number that is legal according to the WUSB
    spec.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index 090ac30..ed5abe8 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -91,7 +91,8 @@
 #include "wusbhc.h"
 
 enum {
- WA_SEGS_MAX = 255,
+ /* [WUSB] section 8.3.3 allocates 7 bits for the segment index. */
+ WA_SEGS_MAX = 128,
 };
 
 enum wa_seg_status {
@@ -588,7 +589,7 @@ static ssize_t __wa_xfer_setup_sizes(struct wa_xfer *xfer,
    xfer->segs = 1;
  }
 
- if (xfer->segs >= WA_SEGS_MAX) {
+ if (xfer->segs > WA_SEGS_MAX) {
   dev_err(dev, "BUG? oops, number of segments %zu bigger than %d\n",
    (urb->transfer_buffer_length/xfer->seg_size),
    WA_SEGS_MAX);
-- 
1.7.1